doc/make.py

```diff
@@ -1,438 +1,423 @@
-    with open(cred, 'r') as fh:
-        server, port, un, domain = fh.read().split(',')
-    port = int(port)
-    login = un + '@' + domain + '.com'
-
-    import base64
-    with open('%s/cron_email_pwd' % tmp_dir, 'r') as fh:
-        pwd = base64.b64decode(fh.read())
-
-    return server, port, login, pwd
-
-
-def _get_config():
-    tmp_dir = _get_dir()
-    with open('%s/addresses' % tmp_dir, 'r') as fh:
-        from_name, to_name = fh.read().split(',')
-    return from_name, to_name
-
-funcd = {
-    'html': html,
-    'zip_html': zip_html,
-    'upload_dev': upload_dev,
-    'upload_stable': upload_stable,
-    'upload_dev_pdf': upload_dev_pdf,
-    'upload_stable_pdf': upload_stable_pdf,
-    'latex': latex,
-    'latex_forced': latex_forced,
-    'clean': clean,
-    'auto_dev': auto_dev_build,
-    'auto_debug': lambda: auto_dev_build(True),
-    'build_pandas': build_pandas,
-    'all': all,
-}
-
-small_docs = False
-
-# current_dir = os.getcwd()
-# os.chdir(os.path.dirname(os.path.join(current_dir, __file__)))
-
-import argparse
-argparser = argparse.ArgumentParser(description="""
-pandas documentation builder
-""".strip())
-
-# argparser.add_argument('-arg_name', '--arg_name',
-#                    metavar='label for arg help',
-#                    type=str|etc,
-#                    nargs='N|*|?|+|argparse.REMAINDER',
-#                    required=False,
-#                    #choices='abc',
-#                    help='help string',
-#                    action='store|store_true')
-
-# args = argparser.parse_args()
-
-#print args.accumulate(args.integers)
+            step = 'clean'
+            cls.clean()
+            step = 'html'
+            cls.html()
+            step = 'upload dev'
+            cls.upload_dev()
+            if not debug:
+                cls._sendmail(step)
+            step = 'latex'
+            cls.latex()
+            step = 'upload pdf'
+            cls.upload_dev_pdf()
+            if not debug:
+                cls._sendmail(step)
+        except (Exception, SystemExit) as inst:
+            msg = str(inst) + '\n'
+            cls._sendmail(step, '[ERROR] ' + msg)
+
+    @classmethod
+    def auto_debug(cls):
+        """Build and upload HTML and PDF documentation with success emails."""
+        cls.auto_dev(debug=True)
+
+    @classmethod
+    def build_pandas(cls):
+        """Compile pandas cython extension."""
+        os.chdir('..')
+        cls._run_os('python',
+                    'setup.py',
+                    'clean')
+        cls._run_os('python',
+                    'setup.py',
+                    'build_ext',
+                    '--inplace')
+        os.chdir('doc')
 
-def generate_index(api=True, single=False, **kwds):
-    from jinja2 import Template
-    with open("source/index.rst.template") as f:
-        t = Template(f.read())
+    @classmethod
+    def all(cls):
+        """Alias to build HTML by default."""
+        cls.html()
+
+    @classmethod
```

Comment:
do we actually support this?

Reply:
the current script does, both `python make.py` and `python make.py all` are equivalent to `python make.py html`. Would rather  get rid of `all` and would show the usage if no command is provided, but I tried to keep compatibility with the old script.

Reply:
i understand but let's remove what we dont need instead